### PR TITLE
Configuration element for alignment scoring

### PR DESCRIPTION
**Pull request by joergi-w**
solves issue #326

### CODE REVIEW

File: include/seqan3/alignment/configuration/align_config_score.hpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+namespace seqan3::detail
+{
+/*!\brief A configuration element for alignment scoring.
+ * \ingroup configuration
+ * \tparam scoring_scheme_type The type of the underlying scoring scheme.
+ */
+template <typename scoring_scheme_type>
+struct align_config_score
+{
+    //! \brief Holds the actual score scores in a score scheme.
```
Comment (rrahn) [inline include/seqan3/alignment/configuration/align_config_score.hpp]: score scores?

File: include/seqan3/alignment/configuration/align_config_score.hpp [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    scoring_scheme_type value;
+};
+
+/*!\brief The score adaptor enabling pipe notation.
+ * \ingroup configuration
+ * \tparam score_config_type A template-template class specifying the actual score implementation.
+ */
+template <template <typename ...> typename score_config_type>
```
Comment (rrahn) [inline include/seqan3/alignment/configuration/align_config_score.hpp]: superfluous tempalte parameter

File: include/seqan3/alignment/configuration/align_config_score.hpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+struct align_config_score_adaptor : public configuration_fn_base<align_config_score_adaptor>
+{
+    /*!\brief Adds to the configuration a score configuration element.
+     * \tparam configuration_type The type of the configuration to be extended.
+     * \tparam scoring_scheme_type The type of the scoring scheme that is included in the new configuration.
+     * \param[in] cfg  The configuration to be extended.
+     * \param[in] scheme The scoring scheme for the algorithm.
+     * \returns A new configuration containing the score configuration element.
+     */
+    template <typename configuration_type, typename scoring_scheme_type>
```
Comment (smehringer) [inline include/seqan3/alignment/configuration/align_config_score.hpp]: there is a `seqan3::scoring_scheme_concept` that should probably be required here

File: include/seqan3/alignment/configuration/align_config_score.hpp [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    {
+        static_assert(is_valid_alignment_configuration_v<align_cfg::id::score, remove_cvref_t<configuration_type>>,
+                      SEQAN3_INVALID_CONFIG(align_cfg::id::score));
+
+        return std::forward<configuration_type>(cfg).push_front(align_config_score<scoring_scheme_type>
+                                                                {std::move(scheme)});
+    }
+};
+
+/*!\brief Helper template meta-function associated with detail::align_config_score.
```
Comment (smehringer) [inline include/seqan3/alignment/configuration/align_config_score.hpp]: link only works with seqan3::

File: include/seqan3/alignment/configuration/align_config_score.hpp (hunk 3 above)
Comment (rrahn) [reply include/seqan3/alignment/configuration/align_config_score.hpp]: I'll fix this later.

File: include/seqan3/alignment/configuration/align_config_score.hpp (hunk 4 above)
Comment (rrahn) [reply include/seqan3/alignment/configuration/align_config_score.hpp]: this to

**Pull request merged**